ppa/archive/models.py

```diff
@@ -415,6 +416,114 @@ def populate_fields(self, field_data):
+        #     preceding_character = value[-1:]
+
+        # if preceding_character == ':':
+        #     self.subtitle = marc_record['245']['b'] or ''
+        # NOTE: skipping preceding character check for now
+        field_data['subtitle'] = marc_record['245']['b'] or ''
+            # strip trailing space & slash from subtitle
+        field_data['subtitle'] = field_data['subtitle'].rstrip(' /')
+        # indicator 2 provides the number of characters to be
+        # skipped when sorting (could be 0)
+        try:
+            non_sort = int(marc_record['245'].indicators[1])
+        except ValueError:
+            # at least one record has a space here instead of a number
+            # probably a data error, but handle it
+            # - assuming no non-sort characters
+            non_sort = 0
+
+        # strip whitespace, since a small number of records have a
+        # nonsort value that doesn't include a space after a
+        # definite article.
+        # Also strip punctuation, since MARC only includes it in
+        # non-sort count when there is a definite article.
+        field_data['sort_title'] = marc_record.title()[non_sort:]\
+                .strip(' "[')
+        field_data['author'] = marc_record.author() or ''
+        # remove a note present on some records and strip whitespace
+        field_data['author'] = field_data['author'].replace('[from old catalog]', '').strip()
+        # removing trailing period, except when it is part of an
+        # initial or known abbreviation (i.e, Esq.)
+        # Look for single initial, but support initials with no spaces
+        if field_data['author'].endswith('.') and not \
+        re.search(r'( ([A-Z]\.)*[A-Z]| Esq)\.$', field_data['author']):
+            field_data['author'] = field_data['author'].rstrip('.')
+
+        # field 260 includes publication information
+        if '260' in marc_record:
+            # strip trailing punctuation from publisher and pub place
+            # subfield $a is place of publication
+            field_data['pub_place'] = marc_record['260']['a'] or ''
+            field_data['pub_place'] = field_data['pub_place'].rstrip(';:,')
+            # if place is marked as unknown ("sine loco"), leave empty
+            if field_data['pub_place'].lower() == '[s.l.]':
+                field_data['pub_place'] = ''
+            # subfield $b is name of publisher
+            field_data['publisher'] = marc_record['260']['b'] or ''
+            field_data['publisher'] = field_data['publisher'].rstrip(';:,')
+            # if publisher is marked as unknown ("sine nomine"), leave empty
+            if field_data['publisher'].lower() == '[s.n.]':
+                field_data['publisher'] = ''
+
+            # remove printed by statement before publisher name
+            field_data['publisher'] = re.sub(
+                self.printed_by_re,
+                '',
+                field_data['publisher'],
+                flags=re.IGNORECASE
+            )
```

Comment:
this seems like an easy candidate for extraction into a separate method, if you wanted to placate codefactor. not sure it matters though